src/compiler/transformers/module/module.ts

```diff
@@ -88,11 +88,14 @@ namespace ts {
             addRange(statements, endLexicalEnvironment());
             addExportEqualsIfNeeded(statements, /*emitAsReturn*/ false);
 
+            if (!currentModuleInfo.exportEquals) {
+                append(statements, createUnderscoreUnderscoreESModule());

```

Comment:
@Kovensky  do you mean `__esModule` has to be emitted as a first statement in the output to prevent circular imports?